multicluster/controllers/multicluster/stale_controller.go

```diff
@@ -270,6 +295,59 @@ func (c *StaleResCleanupController) cleanupClusterInfoResourceExport(commonArea
 	return nil
 }
 
+func (c *StaleResCleanupController) cleanupMemberClusterAnnounce() error {
+	clusterSetList := &mcsv1alpha1.ClusterSetList{}
+	if err := c.List(ctx, clusterSetList, &client.ListOptions{}); err != nil {
+		return err
+	}
+	memberClusterAnnounceList := &mcsv1alpha1.MemberClusterAnnounceList{}
+	if err := c.List(ctx, memberClusterAnnounceList, &client.ListOptions{}); err != nil {
+		return err
+	}
+
+	var members []mcsv1alpha1.MemberCluster
+	// The ClusterSet webhook will guarantee that there will be at most one ClusterSet in one Namespace.
+	if len(clusterSetList.Items) == 1 {
+		members = clusterSetList.Items[0].Spec.Members
+	}
+
+	for _, m := range memberClusterAnnounceList.Items {
+		var exist bool
+		var needToDelete bool
+		memberClusterAnnounce := m
+		for _, member := range members {
+			if member.ClusterID == memberClusterAnnounce.ClusterID {
+				exist = true
+				break
+			}
+		}
+		if !exist {
+			klog.InfoS("Cleaning up stale MemberClusterAnnounce. It does not exist in the member cluster list", "MemberClusterAnnounce", klog.KObj(&memberClusterAnnounce))
```

Comment:
But why we delete MemberAnnounce if it is not in the ClusterSet. It is possible leader cluster has not processed the update yet?

I feel we just check timestamp to delete stale members.

Reply:
FIxed. Just check the timestamp and delete the outdated resources.